data_structures/disjoint_set/alternate_disjoint_set.py

```diff
@@ -0,0 +1,51 @@
+"""
+This code implements a disjoint set using Lists 
+with added heuristics for efficiency
+Union by Rank Heuristic and Path Compression
+"""
+class DisjointSet:
+    def __init__(self, set_count):
```

Comment:
Type hints?  Doctests?  `set_count` is not a single count but is a list or other iterator of counts so `set_counts` (with a trailing s) would be a more intuitive variable name.